tools/generator/Java.Interop.Tools.Generator.ObjectModel/NamespaceMapping.cs

```diff
@@ -31,6 +31,21 @@ public void Generate (CodeGenerationOptions opt, GenerationInfo gen_info)
 				// delegate bool _JniMarshal_PPL_Z (IntPtr jnienv, IntPtr klass, IntPtr a);
 				foreach (var jni in opt.GetJniMarshalDelegates ())
 					sw.WriteLine ($"delegate {FromJniType (jni[jni.Length - 1])} {jni} (IntPtr jnienv, IntPtr klass{GetDelegateParameters (jni)});");
+
+				// [SupportedOSPlatform] only exists in .NET 5.0+, so we need to generate a
+				// dummy one so earlier frameworks can compile.
+				if (opt.CodeGenerationTarget == Xamarin.Android.Binder.CodeGenerationTarget.XAJavaInterop1) {
+					sw.WriteLine ("#if !NET5_0_OR_GREATER");
+					sw.WriteLine ("namespace System.Runtime.Versioning {");
+					sw.WriteLine ("    [System.Diagnostics.Conditional(\"NEVER\")]");
+					sw.WriteLine ("    [AttributeUsage(AttributeTargets.Assembly | AttributeTargets.Class | AttributeTargets.Constructor | AttributeTargets.Event | AttributeTargets.Method | AttributeTargets.Module | AttributeTargets.Property | AttributeTargets.Struct, AllowMultiple = true, Inherited = false)]");
+					sw.WriteLine ("    public sealed class SupportedOSPlatformAttribute : Attribute {");
```

Comment:
If *we* are defining this type, in a pre-.NET 5 context, shouldn't this attribute be `internal`, not `public`?

Reply:
Yep, making it `public` causes type conflicts when a pre-.NET 5 `Mono.Android.dll` is consumed by a .NET 5+ Android app.  Marked `internal`.